### PR TITLE
*: Replace internal testutil AssertEqual function

### LINKED ISSUE #13015: Replace internal testutil AssertEqual function in favour of github.com/stretchr/testify/assert.Equals
etcd in tests to assert equality uses mixed https://github.com/stretchr/testify/assert and etcd' own `testutil.AssertEqual` function. This issue suggests using `github.com/stretchr/testify/assert.Equals` everywhere and removing `testutil.AssertEqual` function.

Example of what we want everywhere: https://github.com/etcd-io/etcd/blob/e2d67f2e3bfa6f72178e26557bb22cc1482c418c/server/datadir/datadir_test.go#L12

The goal is that testutil package to remove is https://github.com/etcd-io/etcd/blob/e299264cf3d93c88c69421adf6b994dafac5ec24/client/pkg/testutil/assert.go#L23 and replace with `assert.Equals` where it is called.

See the background for this in https://github.com/etcd-io/etcd/pull/13011#discussion_r635865695.

### PR DESCRIPTION
**Pull request by gpavanb1**
To maintain uniformity in testing methodology, assert.Equals has been used everywhere while replacing testutil.AssertEqual

Fixes #13015

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: tests/integration/clientv3/lease/leasing_test.go [hunk 1]
```diff
@@ -23,8 +23,9 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
```
Comment (lilic) [inline tests/integration/clientv3/lease/leasing_test.go:26]: Nitpick: let's sort this after all of etcd own imports.

File: tests/integration/v2store/store_tag_not_v2v3_test.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ package v2store_test
 import (
 	"testing"
 
+	"github.com/stretchr/testify/assert"
```
Comment (lilic) [inline tests/integration/v2store/store_tag_not_v2v3_test.go:23]: Same nit as above, lets have etcd imports first, new line then this.

**Review by lilic — COMMENTED** [review]
Very cool, lets have tests run and will have a more detailed review afterwards.

Thank you so much for the quick PR!

File: tests/integration/clientv3/lease/leasing_test.go (hunk 1 above)
Comment (gpavanb1) [reply tests/integration/clientv3/lease/leasing_test.go:26]: VSCode is preferring this import order though with testify/assert sitting on top. Every time I save the file with the new import at the bottom, it shifts back up :/

Comment (lilic) [reply tests/integration/clientv3/lease/leasing_test.go:26]: No worries was a nit!

**Comment by lilic** [comment]
Don't think the failures are related to this or any other functional test failures from the first glance https://github.com/etcd-io/etcd/pull/13017/checks?check_run_id=2631047806:

```
stderr: {"level":"warn","ts":1621521902.734585,"caller":"tester/cluster_run.go:366","msg":"restart FAIL","round":0,"case":19,"case-total":22,"error":"rpc error: code = Unknown desc = dial tcp 127.0.0.1:2380: connect: connection refused, rpc error: code = Unknown desc = dial tcp 127.0.0.1:1380: connect: connection refused"}
stderr: {"level":"error","ts":1621521902.7346072,"caller":"tester/cluster_run.go:323","msg":"functional-tester FAIL","round":0,"case":19,"case-total":22,"error":"wait full health error: etcdserver: too many requests (\"127.0.0.1:1379\")","stacktrace":"go.etcd.io/etcd/tests/v3/functional/tester.(*Cluster).failed
	/home/runner/work/etcd/etcd/tests/functional/tester/cluster_run.go:323
go.etcd.io/etcd/tests/v3/functional/tester.(*Cluster).cleanup
	/home/runner/work/etcd/etcd/tests/functional/tester/cluster_run.go:373
go.etcd.io/etcd/tests/v3/functional/tester.(*Cluster).Run
	/home/runner/work/etcd/etcd/tests/functional/tester/cluster_run.go:61
main.main
	/home/runner/work/etcd/etcd/tests/functional/cmd/etcd-tester/main.go:59
runtime.main
	/opt/hostedtoolcache/go/1.16.4/x64/src/runtime/proc.go:225"}
{"level":"fatal","ts":"2021-05-20T14:44:54.084Z","caller":"etcdmain/etcd.go:203","msg":"discovery failed","error":"cannot fetch cluster info from peer urls: could not retrieve cluster information from the given URLs","stacktrace":"go.etcd.io/etcd/server/v3/etcdmain.startEtcdOrProxyV2\n\t/home/runner/work/etcd/etcd/server/etcdmain/etcd.go:203\ngo.etcd.io/etcd/server/v3/etcdmain.Main\n\t/home/runner/work/etcd/etcd/server/etcdmain/main.go:40\nmain.main\n\t/home/runner/work/etcd/etcd/server/main.go:32\nruntime.main\n\t/opt/hostedtoolcache/go/1.16.4/x64/src/runtime/proc.go:225"}
``

**Review by lilic — COMMENTED** [review]
Lets also remove the function https://github.com/etcd-io/etcd/blob/d9c5e1f0a27a3af2007db3dc8259d6b7f536f28a/client/pkg/testutil/assert.go#L23-L38

**Comment by ptabor** [comment]
```
go.etcd.io/etcd/client/pkg/v3/testutil: package github.com/stretchr/testify/assert imported from implicitly required module; to add missing requirements, run:
```

```
cd client/pkg
go get github.com/stretchr/testify
```

should fix it

**Comment by gpavanb1** [comment]
> ```
> go.etcd.io/etcd/client/pkg/v3/testutil: package github.com/stretchr/testify/assert imported from implicitly required module; to add missing requirements, run:
> ```
> 
> ```
> cd client/pkg
> go get github.com/stretchr/testify
> ```
> 
> should fix it

It looks like there are some issues with `go.sum` entries even after running this. I've tried running `go get -t .` and `go mod tidy`, both of which are returning without change. Would be awesome if you can suggest a way forward!

**Comment by lilic** [comment]
@gpavanb1 what version of go are you using?

**Comment by ptabor** [comment]
`./scripts/fix.sh` should refresh the go.sums

**Comment by gpavanb1** [comment]
I am using 1.15.6 darwin/amd64 for the development.

I have created another commit after running `fix.sh`

**Comment by gpavanb1** [comment]
There seem to be multiple errors in the form
```
    store_ttl_test.go:321: 
        	Error Trace:	assert.go:34
        	            				store_ttl_test.go:321
        	Error:      	Not equal: 
        	            	expected: <nil>(<nil>)
        	            	actual  : *v2store.Event((*v2store.Event)(nil))
        	Test:       	TestStoreRecoverWithExpiration
```

Perhaps the library is testing `nil` in a different fashion compared to previous?

**Comment by ptabor** [comment]
Maybe worth using: `assert.Nil(t, value, msg)` instead

**Comment by ptabor** [comment]
Thank you !

**Comment by lilic** [comment]
Thank you! 🎉

**Pull request merged**